### ISSUE TITLE
Issue loading many regex'es

### ISSUE THREAD

**Issue by iuyiuy** [ingydotnet/yaml-libyaml-pm#64]
This might be related to issue #14, but i'm not sure, since i wasn't able to reproduce it with the command given in that bug report.

I managed to make a test case which consistently worked when creating 125 regex'es and didn't work with 126 (keep in mind that i ran `set +H` in bash, so the exclamation marks will work as intended here):
```
kbj@kbj-dev ~ $ for i in `seq 1 125`; do echo -e "$i:\n  asd: !!perl/regexp (?^u:^qweqwe$)"; done > /tmp/test.yaml; perl -MData::Dumper -MYAML::XS -e 'die Dumper(YAML::XS::LoadFile("/tmp/test.yaml"))' 2>&1 | head
$VAR1 = {
          '123' => {
                     'asd' => qr/(?^u:^qweqwe$)/u
                   },
          '52' => {
                    'asd' => qr/(?^u:^qweqwe$)/u
                  },
          '112' => {
                     'asd' => qr/(?^u:^qweqwe$)/u
                   },
kbj@kbj-dev ~ $ for i in `seq 1 126`; do echo -e "$i:\n  asd: !!perl/regexp (?^u:^qweqwe$)"; done > /tmp/test.yaml; perl -MData::Dumper -MYAML::XS -e 'die Dumper(YAML::XS::LoadFile("/tmp/test.yaml"))' 2>&1 | head
panic: memory wrap at -e line 1.
```

This is with perl v5.20.2, if that makes a difference. I was also able to make a yaml file with 123k regex'es, which apparently worked:
```
kbj@kbj-dev ~ $ perl -MData::Dumper -MYAML::XS -e 'die Dumper(YAML::XS::Load(join("",map {$_.":\n asd:\n  !!perl/regexp (?^u:^qweqwe\$)\n"} 1..123000)))' 2>&1 | head
$VAR1 = {
          '68035' => {
                       'asd' => qr/(?^u:^qweqwe$)/u
                     },
          '82639' => {
                       'asd' => qr/(?^u:^qweqwe$)/u
                     },
          '100609' => {
                        'asd' => qr/(?^u:^qweqwe$)/u
                      },
```

**Comment by perlpunk** [comment]
Thanks for the report.
Reproduced on 5.24.1

**Comment by perlpunk** [comment]
This should be fixed by PR #71
Thanks again!

**Comment by perlpunk** [comment]
Released YAML::LibYAML 0.67_001
